pkgs/build-support/trivial-builders.nix

```
@@ -313,27 +313,27 @@ rec {
    * as a easy way to build multiple derivations at once.
    */
   symlinkJoin =
-    args_@{ name
-         , paths
+    args_@{ paths
          , preferLocalBuild ? true
          , allowSubstitutes ? false
          , postBuild ? ""
          , ...
          }:
     let
-      args = removeAttrs args_ [ "name" "postBuild" ]
-        // {
-          inherit preferLocalBuild allowSubstitutes;
-          passAsFile = [ "paths" ];
-        }; # pass the defaults
-    in runCommand name args
-      ''
-        mkdir -p $out
-        for i in $(cat $pathsPath); do
-          ${lndir}/bin/lndir -silent $i $out
-        done
-        ${postBuild}
-      '';
+      defaultAttr = {
+        inherit preferLocalBuild allowSubstitutes;
+        dontUnpack = true;
+        passAsFile = [ "paths" ];
+        installPhase = ''
```

This should keep semantics as similar to before as possible, specifically we want to avoid that specifying ordinary phases can work (this would be quite confusing, since `postBuild` is executed in `installPhase` here for example), also running things like `fixupPhase` and `configurePhase` as well as `buildPhase` which then no op is unnecessary.

```suggestion
        passAsFile = [ "paths" ];
        buildCommand = ''
```